File: ProductOfArrayExceptSelf/ProductOfArrayExceptSelf.py
class Solution(object):
    def productExceptSelf(self, nums):
        """
        :type nums: List[int]
        :rtype: List[int]
        """
        res = [1]*len(nums)
        left = 1
        if len(nums) < 2:
            return nums
        for i in range(len(nums)-1):
            left *= nums[i]
            res[i+1] *= left
        right = 1
        for i in range(len(nums)-1, 0, -1):
            right *= nums[i]
            res[i-1] *= right
        return res

File: ProductOfArrayExceptSelf/ProductOfArrayExceptSelf.cpp
#include <vector>

using namespace std;

class Solution {
public:
    vector<int> productExceptSelf(vector<int>& nums) {
        vector<int> result(nums.size(), 1);
        int left = 1, right = 1;
        if(nums.size() < 2)
        	return nums;
        for(int i = 0;i<nums.size() - 1;i++) {
        	left *= nums[i];
        	result[i+1] *= left;
        }
        for(int j = nums.size() - 1;j > 0;j--) {
        	right *= nums[j];
        	result[j - 1] *= right;
        }
        return result;
    }
};